client/src/map/shareLocationModal.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import styled from 'styled-components';
+import Clipboard from 'react-clipboard.js';
+import { GoClippy, GoBroadcast } from 'react-icons/go';
+import { FaBroadcastTower } from 'react-icons/fa';
+import Modal, { ModalHeader, ModalParagraph } from '../common/modal';
+
+/**
+ * TODO: This URL should probably be an environment variable
+ */
+const baseUrl = 'http://localhost:8080';
+
+const FlexContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+const IconColumn = styled.div`
+  /* ... */
+`;
+
+const ContentColumn = styled.div`
+  /* ... */
+`;
+
+const BroadcastdIcon = styled.div`
+  height: auto;
+  width: 100px;
+  padding: 30px 15px 0 0;
+
+  & > svg {
+    height: 100%;
+    width: 100%;
+  }
+`;
+
+const ModifiedModalHeader = styled(ModalHeader)`
+  margin-bottom: 10px;
+`;
+
+const HighlightedParagraph = styled(ModalParagraph)`
+  color: red;
+`;
+
+/**
+ * Credit: UrlInput and CopyButton are based on similar components on the GitHub web page
+ */
+const UrlInput = styled.input`
+  width: 80%;
+  font-size: 12px;
+  line-height: 20px;
+  padding: 3px 5px;
+  border: 1px solid #d1d5da;
+  border-radius: 3px 0 0 3px;
+  color: #24292e;
+`;
+
+const ButtonGroup = styled.span`
+  /* ... */
+`;
+
+const CopyButton = styled(Clipboard)`
+  font-size: 12px;
+  line-height: 20px;
+  padding: 3px 10px;
+  margin-left: -1px;
+  border: 1px solid rgba(27, 31, 35, 0.2);
+  border-radius: 0 3px 3px 0;
+  background-color: #eff3f6;
+  cursor: pointer;
+`;
+
+const CopyIcon = styled(GoClippy)`
+  /**
+   * This doesn't seem to work...
+   * The icon is not centered vertically - it's a little off.
+   */
+  display: inline-block;
+  vertical-align: middle;
+  width: 14px;
+  height: 16px;
+`;
+
+interface Props {
+  isOpen: boolean;
+  onClose(): void;
+  currentBluetoothName: null | string;
+}
+
+const ShareLocation = ({ isOpen, onClose, currentBluetoothName }: Props) => {
+  const shareLink = `${baseUrl}/?track=${currentBluetoothName}`;
```

Comment:
It's probably good idea to use the installed query-string module for creating the query string here, we might get bugs if bluetooth name has non-url safe characters.

Reply:
something like `queryString.stringify({track: currentBluetoothName})`, it serializes the values under the hood.